### ISSUE TITLE
bug: Apple Keychain not detecting domain properly, password generation doesn't get triggered

### ISSUE THREAD

**Issue by WetCouch** [ionic-team/capacitor#5633]
### Capacitor Version

```
@capacitor/cli: 3.4.3
@capacitor/core: 3.5.1
@capacitor/android: 3.5.1
@capacitor/ios: 3.4.3
```

### Platform(s)

 iOS

### Current Behavior

When trying to sign in, upon focusing on either username or password fields, Keychain doesn't detect the valid domain (`apple-app-site-association` is served correctly, and the Xcode project is configured to utilise `webcredentials`) and doesn't propose saved passwords for it. If new credentials provided, it doesn't ask to save them in Keychain. When trying to sign up, it doesn't propose a new, strong password in password fields (`autocomplete="new-password"` is added). All of that correctly works in the browser, but not in the Capacitor app.

### Expected Behavior

When trying to sign in, upon focusing on either username or password fields, Keychain should detect the valid domain and propose saved passwords for it. If new credentials provided, it should ask to save them in Keychain. When trying to sign up, it should propose a new, strong password in password fields, as it works in the browser.

### Code Reproduction
Basically any sign-in inputs for not working domain detection:
```html
<input type="email" autocomplete="username" />
<input type="password" autocomplete="current-password" />
```

Any sign-up inputs for not working password generation:
```html
<input type="email" autocomplete="email" />
<input type="password" autocomplete="new-password" />
<input type="password" autocomplete="new-password" />
```

### Other Technical Details

`npm --version` output: `8.3.1`

`node --version` output: `v16.14.0`

`pod --version` output (iOS issues only): `1.11.3`

### Additional Context

`apple-app-site-association` is served correctly, and the Xcode project is configured to utilise `webcredentials`, Apple Validator confirmed its validity.

**Comment by Endrzei** [comment]
I'm facing, what seems to be the same exact issue. I've verified my `apple-app-site-association` with the Apple validator and I've got the `applinks` to work properly, yet I can't get the keychain to detect the association.

**Comment by WetCouch** [comment]
Any updates?

**Comment by jcesarmobile** [comment]
We have verified that it works as expected and created this guide with the required configurations to make it work
https://capacitorjs.com/docs/guides/autofill-credentials

Make sure you are not missing any steps
